### ISSUE TITLE
Rename Roll Mode to Message Mode (or to Message Visibility)

### ISSUE THREAD

**Issue by mkahvi** [foundryvtt/foundryvtt#8856]
### User Experience

The modes affect more than just rolls. Although the chat log input currently uses the option only if you roll there (via `/r` command), programmatic messages will accept any regardless if a roll is involved and render correctly with that. This would mean renaming rollMode in the data also to mode or visibility similarly (eventually).

The chatlog should similarly respect the mode selection regardless if roll is made for easy GM whispering, self messaging and similar, so players and GM alike have easy access to variable message generation without having to resort to chat commands when there's a clear selection box already. This could be a default off option even.

**Comment by aaclayton** [comment]
I agree we are using this concept for more than just rolls now and should probably rename accordingly (with some backwards compatibility). This might be something we can take in V11, or it might need to wait for a later major version.

**Comment by mkahvi** [comment]
Blind roll handling should be extended to non-rolls with this, so people can do blind messages.

**Comment by aaclayton** [comment]
We acknowledge that this is a blemish in the API design that we should fix. This requires more effort than we can justify for V12 though so it will remain as-is and we will re-assess in V13.
